tests/core/test_data.py

```diff
@@ -0,0 +1,153 @@
+"""Unit tests for core.data module."""
+import os
+import shutil
+import tempfile
+import unittest
+
+import relentless
+
+class test_Directory(unittest.TestCase):
+    """Unit tests for core.data.Directory."""
+    def test_init(self):
+        """Test basic creation and methods of Directory."""
+        f = tempfile.TemporaryDirectory()
+
+        cwd = os.getcwd()
+
+        #test creation with existing path
+        d = relentless.data.Directory(f.name)
```

Comment:
Is the import structure of `core` allowing this? I'm not sure if we really want this... as in our other tests, let's design this to test the way we want it to be used:
```py
relentless.Directory()
```
or
```py
relentless.core.Directory()
```
I will open an issue to look into how we do the `core` imports.